podvm/Dockerfile.podvm_builder.rhel

```diff
@@ -8,31 +7,53 @@
 #
 FROM registry.access.redhat.com/ubi9/ubi:9.2
 
-ARG GO_VERSION="1.20.10"
-ARG PROTOC_VERSION="3.11.4"
+ARG ARCH="s390x"
+ARG PACKER_VERSION="v1.9.4"
+ARG PLATFORM="Linux"
+ARG GO_VERSION="1.21.3"
+ARG PROTOC_VERSION="24.3"
```

Comment:
Does bumping these versions is necessary for s390? i suppose it's better to be aligned with the versions described in versions.yaml (maybe we can bump for all if needed)
also is the protoc version correct? sounds strange

Reply:
Oh, I see, so for both golang and portobuf versions we'll have to either:
* bump for all the project including other Dockerfiles and versions.yaml (and hope it won't break anything)
* leave it with the previous versions and add a comment like `# s390 requires golang 1.21.3 and protoc 24.3` and let users to overwrite it at runtime

Reply:
I think the right approch will be to add a default dummy link as done for x86 see [here](https://github.com/confidential-containers/cloud-api-adaptor/blob/d60ce9236298a616900f3c481f9d839f4991eac0/versions.yaml#L19) and [here](https://github.com/confidential-containers/cloud-api-adaptor/blob/d60ce9236298a616900f3c481f9d839f4991eac0/Makefile.defaults#L29)
(which is https://access.redhat.com/downloads/content/433/ver=/rhel---9/9.3%20Beta/s390x/product-software & d436ffeac7d468f75526ca8b8bab1cba64ba3ab2889ccd11fa107a7323b267db is suppose)

Reply:
right, I have updated as such